client/allocrunner/task_coordinator.go

```diff
@@ -0,0 +1,386 @@
+package allocrunner
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// taskCoordinatorState represents a state of the task coordinatator FSM.
+type taskCoordinatorState string
+
+const (
+	taskCoordinatorStateInit      taskCoordinatorState = "init"
+	taskCoordinatorStatePrestart                       = "prestart"
+	taskCoordinatorStateMain                           = "main"
+	taskCoordinatorStatePoststart                      = "poststart"
+	taskCoordinatorStatePoststop                       = "poststop"
```

Comment:
the "only set the type once" trick only works with `iota`, you have to set each of these as `taskCoordinatorState` to actually be that type

https://go.dev/play/p/gIyL3sfdiAw

Reply:
:old_man_yells_at_gohper: